logic/transaction.js

```diff
@@ -15,6 +15,15 @@ var self, modules, __private = {}, genesisblock = null;
 
 __private.types = {};
 
+// Private methods
+__private.calculateFee = function (trs, sender) {
+	var fee = 0;
+	if ( trs.data ) {
+		fee = constants.fees.data;
+	}
+	return __private.types[trs.type].calculateFee.call(this, trs, sender) + fee || false;
```

Comment:
@karmacoma, based on my understanding, previously transaction fees were charged based on the transaction types. But now we are charging a fee if a standardized field (which will be present with all the transaction types) has some value. It seems a bit odd to me, or am I missing something?